frontend/pages/admin/UserManagementPage/components/UserForm/UserForm.tsx

```diff
@@ -93,32 +104,59 @@ const UserForm = ({
   sesConfigured,
   canUseSso,
   isSsoEnabled,
+  isTwoFactorAuthenticationEnabled,
   isApiOnly,
-  isNewUser,
+  isNewUser = false,
   isInvitePending,
   serverErrors,
-  createOrEditUserErrors,
+  addOrEditUserErrors,
   isUpdatingUsers,
-}: ICreateUserFormProps): JSX.Element => {
+}: IAddUserFormProps): JSX.Element => {
+  // For scrollable modal
+  const [isTopScrolling, setIsTopScrolling] = useState(false);
+  const topDivRef = useRef<HTMLDivElement>(null);
+  const checkScroll = () => {
+    if (topDivRef.current) {
+      const isScrolling =
+        topDivRef.current.scrollHeight > topDivRef.current.clientHeight;
+      setIsTopScrolling(isScrolling);
+    }
+  };
+
   const { renderFlash } = useContext(NotificationContext);
 
-  const [errors, setErrors] = useState<any>(createOrEditUserErrors);
-  const [formData, setFormData] = useState<any>({
+  const [errors, setErrors] = useState<IUserFormErrors>({
+    name: null,
+    email: null,
+    password: null,
+  });
+  const [formData, setFormData] = useState<IFormData>({
     email: defaultEmail || "",
     name: defaultName || "",
     newUserType: isNewUser ? NewUserType.AdminCreated : null,
-    password: null,
-    sso_enabled: isSsoEnabled,
+    password: "",
+    sso_enabled: isSsoEnabled || false,
+    two_factor_authentication_enabled:
+      isTwoFactorAuthenticationEnabled || false,
     global_role: defaultGlobalRole || null,
     teams: defaultTeams || [],
     currentUserId,
   });
 
   const [isGlobalUser, setIsGlobalUser] = useState(!!defaultGlobalRole);
 
+  // For scrollable modal (re-rerun when formData changes)
   useEffect(() => {
-    setErrors(createOrEditUserErrors);
-  }, [createOrEditUserErrors]);
+    checkScroll();
+    window.addEventListener("resize", checkScroll);
+    return () => window.removeEventListener("resize", checkScroll);
+  }, [formData]);
+
+  useEffect(() => {
+    if (addOrEditUserErrors) {
+      setErrors(addOrEditUserErrors);
+    }
+  }, [addOrEditUserErrors]);
```

Comment:
Since `addOrEditUserErrors` is a prop passed in, whenever it changes the component will rerender, so
can't we just nix this `useEffect` and set `addOrEditUserErrors` as the default `errors` state?

Reply:
Will do, just gotta rearrange some undefined types to not be undefined.